resources/js/Pages/Results.vue

```diff
@@ -289,42 +292,37 @@
                 // eslint-disable-next-line @typescript-eslint/no-explicit-any
                 const confirmationDialog = this.$refs.confirmation as any;
 
-                this.submitting = true;
-
-                this.showSubmitConfirmation(item);
-
-                // remove decision from this.decisions after it has been
-                // sent to the server successfully, to avoid sending them twice
-                delete this.decisions[item];
-                // we can't access the body tag from inside vue, because the inertia instance
-                // is declared inside it, so we call it from the DOM directly
-                document.body.classList.add('noscroll');
-
                 // use axios in order to preserve saved mismatches
                 try {
                     await axios.put('/mismatch-review', this.decisions[item]);
+
+                    this.showSubmitConfirmation(item);
+
                     // remove decision from this.decisions after it has been
                     // sent to the server successfully, to avoid sending them twice
                     delete this.decisions[item];
 
-                    if(!this.disableConfirmation){
-                        setTimeout(() => { 
-                            confirmationDialog.show();
-                        // the transition between the loading state 
-                        // and the dialog looks better with a small delay between them    
-                        }, SUBMITTING_DELAY_TIME + 100)}
+                    // adding this delay because when the response from the request happens
+                    // too fast the overlay and progressbar flash
+                    setTimeout(() => {
+                        this.submitting = false;
+                        document.body.classList.remove('noscroll');
 
+                        if(!this.disableConfirmation){
+                            confirmationDialog.show();
+                        }
+                    }, 250);
```

Comment:
This is the main reason why I think the overlay should get it's own component. Instead of repeating this wait each time, we can encapsulate this within the component itself with clever usage of props and events.

Reply:
250 is also fine. I chose 1s as an arbitrary number. Why not put this in a variable if we are using it in more than one place, though?

Reply:
Because this is all going to go to a new component I'm working on now.

Reply:
ah, alrighty. :)